lsp/nls/src/usage.rs

```diff
@@ -153,33 +153,39 @@ impl UsageLookup {
 
                         TraverseControl::ContinueWithScope(new_env)
                     }
-                    Term::Let(id, val, body, attrs) => {
+                    Term::Let(bindings, body, attrs) => {
                         let mut new_env = env.clone();
-                        let def = Def::Let {
-                            ident: LocIdent::from(*id),
-                            value: val.clone(),
-                            path: Vec::new(),
-                        };
-                        new_env.insert_def(def.clone());
-                        self.add_sym(def);
-
-                        self.fill(val, if attrs.rec { &new_env } else { env });
+                        for (id, val) in bindings {
+                            let def = Def::Let {
+                                ident: LocIdent::from(*id),
+                                value: val.clone(),
+                                path: Vec::new(),
+                            };
+                            new_env.insert_def(def.clone());
+                            self.add_sym(def);
+                        }
+
+                        for (_, val) in bindings {
+                            self.fill(val, if attrs.rec { &new_env } else { env });
+                        }
```

Comment:
It's probably not going to have any import consequence, but it's a bit annoying that we have to do those double iterations each time just in case the let is recursive. One possibility would be to distinguish both cases in the match, and have a different implementation, which leads obviously to a bit of code duplication. Or adds `if` wherever needed (to do everything in the first `for` and elide the second in the  non-recursive case), but I fear this will make the code not very readable. So I don't have a great idea nor a strong opinion. It's just... itchy :stuck_out_tongue:

Reply:
Yes, I definitely agree. This was the clearest version I could come up with, even though it isn't the most efficient.